### PR TITLE
fix(cli/compile): skip rebundle "bundled" code on compile

### LINKED ISSUE #12670: Skip re-bundling "bundled" code on compile
Deno version: 
```
deno 1.15.3 (release, x86_64-unknown-linux-gnu)
v8 9.6.180.9
typescript 4.4.2
```

Running `deno compile index.js` on a minified/uglified/obfuscated code to be unminified and properly formatted, which is an undesired result. Below are the examples.

Input code:
image

Output binary:
image

### PR DESCRIPTION
**Pull request by ah-yu**
Try to close #12670

### CODE REVIEW

File: cli/main.rs [hunk 1]
```diff
@@ -423,7 +423,18 @@ async fn compile_command(
...
+    let root_module = graph.as_ref().modules()[0];
+    match root_module.media_type {
+      MediaType::Jsx | MediaType::TypeScript | MediaType::Tsx => {
+        bundle_module_graph(graph.as_ref(), &ps, &flags)?.0
+      }
+      _ => root_module.source.to_string(),
+    }
+  } else {
+    bundle_module_graph(graph.as_ref(), &ps, &flags)?.0
+  };
```
Comment (lucacasonato) [inline cli/main.rs:437]: You could do something like this (although format this correctly):

```suggestion
  let source = (graph.as_ref().modules().len() == 1).then(|| {
    let root_module = graph.as_ref().modules()[0];
      match root_module.media_type {
        MediaType::JavaScript => Some(root_module.source.to_string())
        _ => None,
      }
  }).flatten().unwrap_or_else(|| bundle_module_graph(graph.as_ref(), &ps, &flags)?.0);  
```

**Review by lucacasonato — COMMENTED** [review]
Nice test! Just one small comment.

**Comment by ah-yu** [comment]
@lucacasonato updated

**Review by lucacasonato — APPROVED** [review]
Great work, thanks @ah-yu!

**Pull request merged**